testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -8367,23 +8367,15 @@ void checkArrayReverseFunc(SqlOperatorFixture f0, SqlFunction function,
     final SqlOperatorFixture f = f0.withLibrary(SqlLibrary.SPARK);
     f.checkScalar("map_keys(map['foo', 1, 'bar', 2])", "[foo, bar]",
         "CHAR(3) NOT NULL ARRAY NOT NULL");
-    f.checkScalar("map_keys(map['foo', 1, null, 2])", "[foo, null]",
-        "CHAR(3) ARRAY NOT NULL");
+
```

Comment:
can you see who added these tests in the first place and check why they added them? maybe the behavior has changed, or maybe they are not testing spark?

Reply:
@chucheng92 If you have time, could you please review this PR?

Reply:
@mihaibudiu I noticed that the changes here and the changes in the previous [PR](https://github.com/apache/calcite/pull/3787) were made by @chucheng92 in the same jira case. I think it is better to refer to his answer last time.